lib/inspec/file_provider.rb

```diff
@@ -172,9 +172,11 @@ def initialize(path)
 
           @contents[path] = begin # not ||= in a tarball, last one wins
                               res = entry.read
-                              try = res.dup
-                              try.force_encoding Encoding::UTF_8
-                              res = try if try.valid_encoding?
+                              if(res) # not an empty file
```

Comment:
I was initially going to skip this entry at line 166, but I think we want to respect that the file is in the @contents even if it's empty?